compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1107,6 +1154,135 @@ public void visit(BLangSimpleVariable varNode) {
         env.enclPkg.isListenerAvailable |= Symbols.isFlagOn(varNode.symbol.flags, Flags.LISTENER);
     }
 
+    @Override
+    public void visit(BLangUserDefinedType userDefinedType) {
+        if (Types.getReferredType(userDefinedType.getBType()).getKind() == TypeKind.RECORD) {
+            createNewTypeDescInst(userDefinedType.getBType());
```

Comment:
Why do we need to do this? When do we get here?

Reply:
Removed this logic as it is no longer needed with recent changes in the implementation(earlier this was needed because In the desugaring phase, we created a user-defined type for the local record).